src/routes/safe/store/actions/fetchTransactions.js

```diff
@@ -245,11 +255,16 @@ export const loadSafeTransactions = async (safeAddress: string) => {
     console.error(`Requests for outgoing transactions for ${safeAddress} failed with 404`, err)
   }
 
-  const txsRecord = await Promise.all(
+  const txsRecord: Array<RecordInstance<TransactionProps>> = await Promise.all(
     transactions.map((tx: TxServiceModel) => buildTransactionFrom(safeAddress, tx)),
   )
 
-  return Map().set(safeAddress, List(txsRecord))
+  const groupedTxs = List(txsRecord).groupBy((tx) => (tx.get('cancellationTx') ? 'cancel' : 'outgoing'))
+
+  return {
+    outgoing: Map().set(safeAddress, groupedTxs.get('outgoing')),
```

Comment:
why not `Map({ safeAddress: groupedTxs.get('outgoing')})`?

Reply:
Because the key is the `safeAddress` value. Using your example, it would have to be: `Map({ [safeAddress]: groupedTxs.get('outgoing') })` (look at the square brackets surrounding `safeAddress`)

I think that the current (`Map().set(...)`) goes more along with immutable style.

But if there's a performance penalty or anything that I'm not seeing by using the current, I have no problem in changing it.

Reply:
yeah actually .set is faster :O
https://www.measurethat.net/Benchmarks/Show/7060/0/immutable-js-initial-value-vs-set